cmd/fs-v1.go

```diff
@@ -512,6 +514,48 @@ func (fs *FSObjects) GetObject(ctx context.Context, bucket, object string, offse
 		return err
 	}
 	defer objectLock.RUnlock()
+
+	if !isCompressed(objInfo.UserDefined) {
```

Comment:
Object layer shouldn't have this information @Praveenrajmani , you should move this code to the handler.

Reply:
I need the have the object locked before doing this ? how do i do this in handlers ? @harshavardhana

Reply:
I particularly added this to preserve the read-write rule.

Reply:
There is a way.. give me some time to think about it.